### PR TITLE
bugfix/340

### LINKED ISSUE #340: Aircraft not reaching targeted speed
Sometimes aircraft will get "stuck" at a speed that is less than 0.1 knots away from the target speed, and not be updated, due to the fact that we were explicitly allowing a small level of error in the code. Really though, that is not needed, and should be removed to prevent cases like this.

This is the reason on a rare occasion, aircraft will start their takeoff roll, _appear_ to reach their rotation speed, and then just never rotate. It's because they rotate at 130, and are remaining at 129.994303546375. If the aircraft happens to calculate that speed between physics updates, they would end up getting stuck there forever.

Need to change this so aircraft will insist upon reaching their exact targeted speed.

### PR DESCRIPTION
**Pull request by erikquinn**
Resolves #340.

Ensures aircraft reach targeted speeds.

### CODE REVIEW

File: src/assets/scripts/client/aircraft/AircraftInstanceModel.js [hunk 1]
```diff
@@ -1558,28 +1558,29 @@ export default class AircraftInstanceModel {
      * @method updateWarning
      */
     updateSpeedPhysics() {
-        let difference = null;
+        let speedChange = 0;
+        const differenceBetweenPresentAndTargetSpeeds = this.speed - this.target.speed;
+        const decelerationFactorDueToGroundBraking = 3.5;
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/AircraftInstanceModel.js]: Should this list be in a constants file?

File: src/assets/scripts/client/aircraft/AircraftInstanceModel.js [hunk 2]
```diff
@@ -1558,28 +1558,29 @@ export default class AircraftInstanceModel {
...
 
-        if (this.target.speed < this.speed - 0.01) {
-            difference = -this.model.rate.decelerate * window.gameController.game_delta() / 2;
+        if (differenceBetweenPresentAndTargetSpeeds === 0) {
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/AircraftInstanceModel.js]: So much cleaner 👍

File: src/assets/scripts/client/aircraft/AircraftInstanceModel.js [hunk 3]
```diff
@@ -1558,28 +1558,29 @@ export default class AircraftInstanceModel {
...
-                // What is 3.5 is this restiance/breaking power?
-                difference *= 3.5;
+                speedChange *= decelerationFactorDueToGroundBraking;
             }
-        } else if (this.target.speed > this.speed + 0.01) {
-            difference  = this.model.rate.accelerate * window.gameController.game_delta() / 2;
-            difference *= extrapolate_range_clamp(0, this.speed, this.model.speed.min, 2, 1);
+        } else if (this.speed < this.target.speed) {
+            speedChange  = this.model.rate.accelerate * window.gameController.game_delta() / 2;
+            speedChange *= extrapolate_range_clamp(0, this.speed, this.model.speed.min, 2, 1);
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/AircraftInstanceModel.js]: What does the 1 and 2 mean?

File: src/assets/scripts/client/aircraft/AircraftInstanceModel.js [hunk 4]
```diff
@@ -1558,28 +1558,29 @@ export default class AircraftInstanceModel {
...
-
-            if (abs(offset) < abs(difference)) {
-                this.speed = this.target.speed;
-            } else {
-                this.speed += difference;
-            }
+        if (abs(speedChange) > abs(differenceBetweenPresentAndTargetSpeeds)) {
+            this.speed = this.target.speed;
+        } else {
+            this.speed += speedChange;
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/AircraftInstanceModel.js]: Could we remove the else by moving this assignment above the if and then overriding it with the existing if?

```
this.speed += speedChange;

If (abs(speedChange) > abs(differenceBetweenPresentAndTargetSpeeds)) {
    this.speed = this.target.speed;
}

File: src/assets/scripts/client/aircraft/AircraftInstanceModel.js (hunk 1 above)
Comment (erikquinn) [inline src/assets/scripts/client/aircraft/AircraftInstanceModel.js]: Eh, yeah. I was being lazy. :smile: 

Fixed.

File: src/assets/scripts/client/aircraft/AircraftInstanceModel.js (hunk 3 above)
Comment (erikquinn) [inline src/assets/scripts/client/aircraft/AircraftInstanceModel.js]: Honestly couldn't really make sense of it. Seems backwards, but obviously it's right. Some convoluted solution that's above my head at the moment.

File: src/assets/scripts/client/aircraft/AircraftInstanceModel.js (hunk 4 above)
Comment (erikquinn) [inline src/assets/scripts/client/aircraft/AircraftInstanceModel.js]: Sure, I guess.

Fixed.

**Pull request merged**